hmsc/updaters/updateZ.py

```diff
@@ -43,7 +43,7 @@ def updateZ(params, data, rLHyperparams, *,
     Pi = data["Pi"]
     distr = data["distr"]
 
-    if len(X.shape.as_list()) == 2: #tf.rank(X) X.ndim == 2:
```

Comment:
Have you tested this in Graph execution mode in GPU?

Reply:
Yes, this works at least with TF 2.12.